### DISCUSSION TITLE
Does GEI support migrating autolinks/permalinks when a repo changes orgs?

### DISCUSSION THREAD

**Discussion by thorntonryan** [github/gh-gei#809] [Q&A] [answered]
### Question

Does GEI support migrating autolinks/permalinks found in issue/PR comments if a repo is changing orgs?

If not, what approach, if any, is recommended?

### Context

I'm trying to understand the implications for various autolinks when migrating a repo to a different organization.

I'm most interested in the following scenarios commonly found in issue or PR comments.

Below is the transform I'm hoping to achieve (somehow):

#### Cross link reference to issue/PR on other repo within org

```diff
-<old>/<repo>#<num>
+<new>/<repo>#<num>
```

```diff
-https://old.instance/<old_org>/<repo>/issues/<num>
+https://new.instance/<new_org>/<repo>/issues/<num>
```

```diff
-https://old.instance/<old_org>/<repo>/pull/<num>
+https://new.instance/<new_org>/<repo>/pull/<num>
```

#### Cross reference to commit of other repo within org

```diff
-<old>/<repo>@<commit>
+<new>/<repo>@<commit>
```

```diff
-https://old.instance/<old_org>/<repo>/commit/<commit>
+https://new.instance/<new_org>/<repo>/commit/<commit>
```

#### [Permalinks to code snippet](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet) 

```diff
-https://old.instance/old_org/repo/blob/commit/file#L1234-L1235
+https://new.instance/new_org/repo/blob/commit/file#L1234-L1235
```

Thanks for your support!

**Comment by timrogers** [comment]
@thorntonryan Thanks for reaching out! I'm just looking into this with the team, and I'll get back to you soon.

**Reply by thorntonryan** [reply]
Thanks for looking into this! I had forgotten about the RAW form that also can be turned into short links, so I've updated the my question with a few more patterns.

**Comment by timrogers** [comment, marked as answer]
@thorntonryan Thanks for your patience on this! The answer to this question is a little complicated, so I'll break it down step by step:

* If you use the "short permalink" style (e.g. `org/repo#<num>`), and the name of the organization and repo is the same after the migration as it was before the migration, then those links will continue to work.
* If you use the "short permalink" style (e.g. `org/repo#<num>`) or a full URL and you run an organization-to-organization migration (see the [docs](https://docs.github.com/en/early-access/enterprise-importer/migrating-organizations-with-github-enterprise-importer/migrating-organizations-from-githubcom-to-github-enterprise-cloud), only available when migrating from GitHub.com to GitHub.com), then we will rewrite the URLs to match your new GitHub host, organization and repo name.
* In all other cases, links will not be re-written during a migration.

I hope that helps! If you have any further questions, or if you think I've made any mistakes, then do feel free to let me know.
